### DISCUSSION TITLE
数据怎么备份保存？

### DISCUSSION THREAD

**Discussion by ayunhe** [usememos/memos#154] [Q&A]
如果要长期使用，应该怎么备份保存数据？

**Comment by lqwakeup** [comment]
从部署者的角度来看，目前系统产生的所有数据会存在 SQLite 的 .db 数据库文件里，可以自行对这个文件进行拷贝，感觉也不太需要额外地进行导出 JSON、备份。

**Comment by ayunhe** [comment]
> 从部署者的角度来看，目前系统产生的所有数据会存在 SQLite 的 .db 数据库文件里，可以自行对这个文件进行拷贝，感觉也不太需要额外地进行导出 JSON、备份。

这个文件在哪里，小白弱弱问一下？

**Comment by lqwakeup** [comment]
看你在启动 docker 时，将哪个本地路径映射到了 `/var/opt/memos`。

---

如果你的启动命令是这样的话：

``` docker
docker run \
  --name memos \
  --publish 5230:5230 \
  --volume ~/.memos/:/var/opt/memos \
  neosmemo/memos:latest \
  --mode prod \
  --port 5230
```

那么该文件就会是 `~/.memos/memos_prod.db`。

**Reply by Run-os** [reply]
大佬，我找不到db文件啊
![image](https://github.com/usememos/memos/assets/77967997/7d600ec6-5771-490d-b998-47285bd6bc32)

我打开的文件路径：/var/lib/docker/containers/5e6cfb32b2421817c34d93b17b85051ea761528dade8ab3386da341b38ac7045

**Reply by Run-os** [reply]
额外说明下，我用了s3，不过db文件s3里面也没有，只看到了图片

**Reply by Run-os** [reply]
抱歉，我在root文件夹下找到了，打扰了
/root/.memos

**Comment by ayunhe** [comment]
> 看你在启动 docker 时，将哪个本地路径映射到了 `/var/opt/memos`。
> 
> 如果你的启动命令是这样的话：
> 
> ```
> docker run \
>   --name memos \
>   --publish 5230:5230 \
>   --volume ~/.memos/:/var/opt/memos \
>   neosmemo/memos:latest \
>   --mode prod \
>   --port 5230
> ```
> 
> 那么该文件就会是 `~/.memos/memos_prod.db`。

谢谢大佬 找到了

**Comment by miaoxiaozhao** [comment]
除了备份数据库，上传的素材如何备份？有指定的目录么？

**Comment by redisread** [comment]
对于部署在无服务器的平台上，大部分是拿不到db文件的，感觉还是需要有导出和导入的功能。😶

**Comment by xudaolong** [comment]
https://github.com/xudaolong/memos-desktop 我做了下离线版本的 下班就用坚果云同步下就好了

**Reply by logicmlx** [reply]
求大佬更新，1.0.0 版本貌似只能选择 S3 存储？能加个本地存储的选项吗，webdav 之类的
![image](https://user-images.githubusercontent.com/52021429/231366594-955aa3ab-5261-4f96-b3e4-b18d89baae39.png)

**Comment by Domonlee** [comment]
我现在是用Git定期的提交db文件

**Reply by HowToSaveMyLife** [reply]
是手动提交么，每次提交都得停止运行吧

**Reply by Domonlee** [reply]
不用的，在服务器写了定时脚本 commit

**Reply by CellLz** [reply]
因为SQlite没法增量备份，所以每次提交git都会产生很大的文件体积，用不了多久硬盘就沾满了。

**Reply by Domonlee** [reply]
资源文件放在R2上，纯文本占不了多少空间。三年了打成zip 1.2M多一点，提交了400多次，目前感觉良好。

**Reply by Domonlee** [reply]
不过你也点了我，再写个定时任务删zip ^_^

**Comment by aoliaoduo** [comment]
请问怎么从memos_prod.db中恢复数据呢?

**Reply by gtxykn0504** [reply]
有一个读取db文件内容的方法，通过DB Browser for SQLite（https://sqlitebrowser.org/ ），将DB文件导入该Browser之后，依次点击【浏然数据】，选择memo的表，就可以看到memo数据了
![image](https://github.com/user-attachments/assets/9fab6cf4-c63e-4554-a3cd-f604f6bb6db4)

**Comment by vibbe** [comment]
Sorry for interrupting, their is a discussion about this export and import function. (English though, hence Chinese isn't my tongue)
Discussion: Backup and import #3376

**Reply by vibbe** [reply]
This is the original discussion Ability to Fully Export and Import All Memos and Instance Configurations [https://github.com/usememos/memos/issues/2000](url) was closed as not planned, unfortunate.
